### ISSUE TITLE
PHPUnit as a dependency

### ISSUE THREAD

**Issue by uuf6429** [php-vcr/phpunit-testlistener-vcr#31]
Since this project is for PHPUnit and [uses PHPUnit classes](https://github.com/php-vcr/phpunit-testlistener-vcr/blob/master/src/VCRTestListener.php#L6-L11), doesn't it explicitly depend on PHPUnit?
In particular, shouldn't depend on a specific PHPUnit version that is compatible?

This causes [an issue](https://github.com/wsdl2phpgenerator/wsdl2phpgenerator/pull/333#issuecomment-409042950) when trying to support multiple versions of PHPUnit. This package does not downgrade to the version that works with PHPUnit.

**Comment by renatomefi** [comment]
Hello @uuf6429, I took the freedom to answer directly the mentioned issue! Hope it helps

**Comment by uuf6429** [comment]
Thanks! 3.1 would work, but only because of the common dependency on PHP. We can't use 3.0 for PHP 7.0 since it [doesn't require it](https://github.com/php-vcr/phpunit-testlistener-vcr/blob/3.0/composer.json#L22).

**Comment by renatomefi** [comment]
If you have ^3.0 it will automatically solve it, by matching all the dependent versions

**Comment by uuf6429** [comment]
Sorry to persist, but ^3.0 doesn't have any requirements... which is incorrect (same with any version, actually).
As it is right now, this package claims to be compatible with any PHPUnit version, even unreleased future ones...

**Comment by renatomefi** [comment]
This library only depends on TestListener (https://github.com/sebastianbergmann/phpunit/blob/master/src/Framework/TestListener.php) and indeed "saying" it's compatible with all versions. Which can be improved for sure.
The real phpunit version limitations are coming from php-vcr/php-vcr and not here.

I'd say we could add a constraint saying it's compatible with php unit 5,6,7. But in the end you'll be limited to the php-vcr one!
Is that what you're looking for?

**Comment by uuf6429** [comment]
> This library only depends on TestListener...
To me that's a clear dependency. If someone installs this project but forgets to install  PHPUnit, s/he end up with a class extending and referencing non-existent classes.

> The real phpunit version limitations are coming from php-vcr/php-vcr and not here.
I do not understand your point here... php-vcr [does not depend](https://github.com/php-vcr/php-vcr/blob/master/composer.json#L20-L23) on phpunit to function, it uses [phpunit only for tests](https://github.com/php-vcr/php-vcr/search?q=phpunit+-path%3A%2Ftests%2F&unscoped_q=phpunit+-path%3A%2Ftests%2F).

> I'd say we could add a constraint saying it's compatible with php unit 5,6,7.
That's the best case scenario. Right now, it doesn't even [suggest](https://getcomposer.org/doc/04-schema.md#suggest) that you should have PHPUnit installed, nor does it say that it [conflicts](https://getcomposer.org/doc/04-schema.md#conflict) with older PHPUnit.

**Comment by renatomefi** [comment]
Yes you're right! php-vcr doesn't depend on it :man_facepalming: 

The 3.1 was done on purpose to provide an easy update path for people using ^3.0, I also agree it isn't the best approach, check more here: https://github.com/php-vcr/phpunit-testlistener-vcr/pull/26

Please feel free to open a PR providing the phpunit dependency, luckily there's no conflicts so far since the mentioned interface remains the same for a long time

**Comment by uuf6429** [comment]
No problem! I'll see if I can get this right for all major versions. Worst case we might end up with a few patch releases.

**Comment by uuf6429** [comment]
@renatomefi I'd like to have two branches, one called `1.1` (branching from tag 1.1.7) and one `2.0` (branching from tag `2.0.0`). When done, I can create my pull requests for both.

**Comment by ekeyte** [comment]
@uuf6429 @renatomefi Hey there, just following up on this issue.

**Comment by uuf6429** [comment]
I'm still waiting for the PRs (linked above) to be merged and released and for branches to the old versions to be created. I literally can't do anything more from my side.

**Comment by ekeyte** [comment]
@uuf6429 @renatomefi Hey guys, are there any updates on this? This is holding us back from moving some big projects to 7.2. Anything I can do to help with this?

**Comment by uuf6429** [comment]
> I literally can't do anything more from my side.
Still applies..

**Comment by ekeyte** [comment]
> > I literally can't do anything more from my side.
> 
> Still applies..

I know, bud. :(

@renatomefi: At your encouragement, @uuf6429  has opened a PR. Is there something holding you back from merging it that I could assist with? We'd love to see this resolved soon. Many thanks!
